franklin/core/tests.py

```diff
@@ -0,0 +1,89 @@
+import os
+from requests.exceptions import ConnectionError, HTTPError, Timeout
+from unittest import mock
+
+from django.test import TestCase
+
+from .helpers import make_rest_get_call, make_rest_post_call
+
+
+class HelpersTestCase(TestCase):
+    def setUp(self):
+        self.url = os.environ['BUILDER_URL'] + '/build'
+        self.headers = {'content-type': 'application/json'}
+        self.body = {
+            "deploy_key": "deploy_key",
+            "branch": "branch",
+            "tag": "tag",
+            "git_hash": "git_hash",
+            "repo_owner": "owner",
+            "path": "path",
+            "repo_name": "name",
+            "environment_id": 1
+            }
+
+    @mock.patch('core.helpers.requests.post', side_effect=ConnectionError)
+    def test_make_rest_post_call_conn_error(self, mock_post):
+        """ Tests make_rest_post_call when the call has an Connection exception.
+        """
+        response = make_rest_post_call(self.url, self.headers, self.body)
+        self.assertEqual(response.status_code, 500)
+
```

Comment:
Now that I know how to mock the request library properly, I went ahead and created a bunch of tests to validate some of our helper functions.

Found a bug in the process.